### PR TITLE
TT-496 Removing disable_management_poller config opt

### PR DESCRIPTION
**Pull request by tbuchaillot**
## Description

Removing `uptime_tests.disable_management_poller` config option.
Now, `uptime_tests.disable` is the trigger of the HostCheckerManager. If it is `true`, the gateway won't be able to become the Host checker manager.

## Related Issue

https://tyktech.atlassian.net/browse/TT-496

## Motivation and Context

https://tyktech.atlassian.net/browse/TT-496

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring or add test (improvements in base code or adds test coverage to functionality)

## Checklist

- [ ] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). If pulling from your own
      fork, don't request your `master`!
- [ ] Make sure you are making a pull request against the **`master` branch** (left side). Also, you should start
      *your branch* off *our latest `master`*.
- [ ] My change requires a change to the documentation.
  - [ ] If you've changed APIs, describe what needs to be updated in the documentation.
  - [ ] If new config option added, ensure that it can be set via ENV variable
- [ ] I have updated the documentation accordingly.
- [ ] Modules and vendor dependencies have been updated; run `go mod tidy && go mod vendor`
- [ ] When updating library version must provide reason/explanation for this update.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] Check your code additions will not fail linting checks:
  - [ ] `go fmt -s`
  - [ ] `go vet`

### CODE REVIEW

File: gateway/server.go [hunk 1]
```diff
@@ -169,11 +169,11 @@ func setupGlobals(ctx context.Context) {
...
+	if !config.Global().UptimeTests.Disable && config.Global().ManagementNode {
 		mainLog.Warn("Running Uptime checks in a management node.")
 	}
-	// Initialise our Host Checker if it's enabled to be a management poller
-	if !config.Global().UptimeTests.DisableManagementPoller {
+	// Initialise HostCheckerManager only if uptime tests are enabled.
+	if !config.Global().UptimeTests.Disable {
 		healthCheckStore := storage.RedisCluster{KeyPrefix: "host-checker:"}
 		InitHostCheckManager(&healthCheckStore)
 	}
```
Comment (gernest) [inline gateway/server.go:179]:
```suggestion
	// Initialise HostCheckerManager 
	if !config.Global().UptimeTests.Disable {
		config.Global().ManagementNode {
			mainLog.Warn("Running Uptime checks in a management node.")
		}
		healthCheckStore := storage.RedisCluster{KeyPrefix: "host-checker:"}
		InitHostCheckManager(&healthCheckStore)
	}
```

**Review by gernest — COMMENTED** [review]
Looks good so far. I left a suggestion.

File: gateway/server.go (hunk 1 above)
Comment (tbuchaillot) [reply gateway/server.go]: good catch! simplified  👍

**Review by gernest — APPROVED** [review]
LGTM

**Comment by gernest** [comment]
@tbuchaillot I missed that this is against `release-2.9` branch. I think It should be against `master` branch but backported to `release-2.9`

**Comment by tbuchaillot** [comment]
> @tbuchaillot I missed that this is against `release-2.9` branch. I think It should be against `master` branch but backported to `release-2.9`

@gernest that config option was only in 2.9 - we didn't go against master since we tried to backporting it from master to release-2.9 with @matiasinsaurralde and had a lot of problems. I'm going to send another PR for master with all the HostCheckerManager changes.

**Comment by buger** [comment]
Ok, and ensure that this new PR will have all the tests.

**Pull request merged**